Tutorials/intro-to-quam/trapped_ion/machine.py

```diff
@@ -0,0 +1,212 @@
+# %%
```

Comment:
I would maybe split this file into 3 files for better readability:

- Quam generation, until line 108
- Single X gate, until line 158 (need to add the loading of QUAM)
- Double X optimization until the end (need to add the loading of QUAM)

What do you think?